CONTRIBUTING.md

```diff
@@ -17,19 +84,219 @@ augroup gno_autocmd
+- `examples` - contains the example `.gno` realms and packages. This is the central point for adding user-defined realms
+  and packages.
+- `gnoland` - contains the base source code for bootstrapping the Gnoland node
+- `pkgs` - contains the dev-audited packages used throughout the gno codebase
+- `stdlibs` - contains the standard library packages used (imported) in `.gno` Smart Contracts. These packages are
+  themselves `.gno` files.
+- `tests` - contains the standard language tests for Gnolang
+
+## How do I?
+
+### How do I submit changes?
+
+Pull Requests serve primarily as an addition to the codebase, but also as a central point of discussion for potential features.
+Instead of opening issues, contributors are encouraged to open PRs with their suggested changes, however small, in order to
+get feedback from the community on their work.
+
+The key to successfully submitting upstream changes is providing the adequate context, with the correct implementation,
+of course.
+
+To let the core team know about _what_ your PR does, and _why_ it does it, you should take a second to fill out one of
+two PR templates provided on the repo.
+
+Once someone leaves a PR review (with open comments / discussions), it is on the PR _creator_ to do their best in trying
+to resolve all comments.
+The PR _reviewer_ is in charge of resolving each conversation (hitting the `Resolve Conversation` button), after the PR
+_creator_ implements or discusses
+the requested changes.
+
+If the PR _creator_ pushed code as a result of some conversation, they should link the commit hash in the relevant
+comment reply, ex:
+
+```Markdown
+Fixed in:
+
+[0a0577c](https://github.com/gnolang/gno/commit/0a0577ccdeb951a6621d6fbe1c04ac4e64a529c1)
+```
+
+#### Base PR template
+
+The [base PR template](https://github.com/gnolang/gno/blob/master/.github/pull_request_template.md) is by default a
+simple template meant to illustrate quickly what’s the context of the PR. Here, you should describe your PR in detail,
+and leave a remark as to how your changes have been tested.
+
+If you've run a manual test, please provide the exact steps taken.
+
+#### Detailed PR template
+
+The [detailed PR template](https://github.com/gnolang/gno/blob/master/.github/PULL_REQUEST_TEMPLATE/detailed_pr_template.md)
+is used when you want to convey a bit more detail and context for your PRs. In contrast to the default PR template, it
+follows a checklist approach that is meant to be filled out by the PR creator, with as much information as possible.
+These detailed PR descriptions serve an important purpose down the line in the future, when it’s important to understand
+the context under which a PR has been created, and for what reason.
+
+You can utilize the detailed PR template by appending the following query param to the URL on the new PR page:
+
+`template=detailed_pr_template.md`
+
+### How do I report a bug?
+
+Found something funky while using gno? You can report it to the team using GitHub issues.
```

Comment:
I know it's obvious, but it's good to mention:
```
Before you open an issue, please check for existing open and closed Issues
to see if that bug/feature has already been reported/requested. If you find 
a relevant topic, you can comment on that issue.
```

Reply:
If I've learned anything, it's that it's _never_ obvious enough 🙂 

Added:

[e1d2213](https://github.com/gnolang/gno/pull/469/commits/e1d22139aca34535ea49c7681ad25d8e15f8651e)